File: new/cpp/150_evaluate_reverse_polish_notation_brute_force.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        // idea: brute force
        // while loop until we have one token left. 
        // for loop each token in the tokens until we find one of these signs: +, -, *, /
        // if we find the sign at ith index, a = value at i - 2, b = value at i - 1
        // find relevant result of the relevant operation: a + b, a - b, a * b, a / b
        // insert the result in the correct position {..., i - 2, <NEW VALUE>, i + 1, ....}
        // return the new tokens
        while (tokens.size() > 1) {
            for (int i = 0; i < tokens.size(); i++) {
                if (tokens[i] == "+" || tokens[i] == "-" || tokens[i] == "*" || tokens[i] == "/") {
                    int a = stoi(tokens[i - 2]);
                    int b = stoi(tokens[i - 1]);
                    int result = 0;
                    if (tokens[i] == "+") {
                        result = a + b;
                    } else if (tokens[i] == "-") {
                        result = a - b;
                    } else if (tokens[i] == "*") {
                        result = a * b;
                    } else {
                        result = a / b;
                    }
                    tokens.erase(tokens.begin() + i - 2, tokens.begin() + i + 1);
                    tokens.insert(tokens.begin() + i - 2, to_string(result));
                    break;
                }
            }
        }
        return stoi(tokens[0]);
    }
    // time: O(n^2)
    // space: O(n)
};

File: new/python/150_evaluate_reverse_polish_notation_brute_force.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        # idea: brute force
        # while loop until we have exactly one element in the given array
        # for loop inside the while loop to iterate through each element and until we hit one of the symbols:
        # "+-*/"
        # if we hit the symbol, then 2 positions before the symbol is a, one position before the symbol is b
        # calculate relevant output, a + b, a - b, a * b, a / b
        # convert it to string, place it in the correct position [:i-2] + [result] + [i+1:], return the array
        while len(tokens) > 1:
            for i in range(len(tokens)):
                if tokens[i] in "+-*/":
                    a = int(tokens[i - 2])
                    b = int(tokens[i - 1])
                    if tokens[i] == "+":
                        result = a + b
                    elif tokens[i] == "-":
                        result = a - b
                    elif tokens[i] == "*":
                        result = a * b
                    elif tokens[i] == "/":
                        result = int(a / b)  # have to do integer division
                    tokens = tokens[: i - 2] + [str(result)] + tokens[i + 1 :]
                    break
        return int(tokens[0])
        # time: O(n^2)
        # space: O(n)
